lib/component-repository/src/routes/virtual-components/GetAction.js

```diff
@@ -1,3 +1,31 @@
+const mongoose = require('mongoose');
+const ComponentVersion = require('../../models/ComponentVersion');
+
 module.exports = async function (req, res, next) {
-    return next();
+  const { componentVersionId, actionName } = req.params;
+
+  if (!mongoose.Types.ObjectId.isValid(componentVersionId)) {
+    return res
+      .status(400)
+      .send({ errors: [{ message: 'Invalid id', code: 400 }] });
+  };
+
+  const componentVersion = await ComponentVersion.findOne(
+    {
+      _id: componentVersionId,
+      'actions.name': actionName,
+    },
+    {
+      'actions.$': 1,
+    }
+  ).lean();
+  if (!componentVersion.actions[0]) {
```

Comment:
Here, I am not sure, but I think that if the component version wouldn't exist, you would get an unhandled error

Reply:
I guess you would check better the componentVersion variable instead of its nested property

Reply:
Done